File: docs/src/testers/sessions/capabilities/captcha_ensure_requirements.py
# @sniptest filename=captcha_ensure_requirements.py
# @sniptest show=1-10
from notte_sdk import NotteClient

client = NotteClient()

# Ensure all requirements are met
with client.Session(
    solve_captchas=True,  # Must be enabled
    proxies=True,  # Helps with detection
) as session:
    pass

status = session.status()
assert status.proxies is True

File: docs/src/testers/sessions/capabilities/captcha_ensure_requirements.ts
// @sniptest filename=captcha_ensure_requirements.ts
// @sniptest show=1-11
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

const session = client.Session({
  solve_captchas: true,
  proxies: true,
});
await session.use(async () => {
  // Use your session here
});

const status = await session.status();
if (status.proxies !== true) throw new Error('Session proxies mismatch');

export { status };
